packages/core/src/sync-store/index.ts

```diff
@@ -845,7 +967,144 @@ export const createSyncStore = ({
+            );
+
+            const endClock = startClock();
+
+            currentTransactionReceiptsRows =
+              await getEventBlockData_transaction_receipts(
+                {
+                  fromBlock,
+                  fromTransactionIndex,
+                  toBlock: supremum,
+                  chainId,
+                  limit,
+                },
+                database,
+              );
+
+            common.metrics.ponder_database_method_duration.observe(
+              { method: "getEventBlockData_transaction_receipts" },
+              endClock(),
+            );
+
+            for (const row of currentTransactionReceiptsRows) {
+              // @ts-ignore
+              transactionReceiptsRows.push(row);
+            }
+          }
+        };
+
+        const logsPromise = async () => {
+          let currentLogsRows = logsRows;
+          while (currentLogsRows.length === limit) {
+            const fromLogIndex = currentLogsRows[limit - 1]!.logIndex;
+            const fromBlock = Number(currentLogsRows[limit - 1]!.blockNumber);
+
+            const endClock = startClock();
+
+            currentLogsRows = await getEventBlockData_logs(
+              {
+                filters: logFilters,
+                fromBlock,
+                fromLogIndex,
+                toBlock: supremum,
+                chainId,
+                limit,
+              },
+              database,
+            );
+
+            common.metrics.ponder_database_method_duration.observe(
+              { method: "getEventBlockData_logs" },
+              endClock(),
+            );
+            for (const row of currentLogsRows) {
+              // @ts-ignore
+              logsRows.push(row);
+            }
+          }
+        };
+
+        const tracesPromise = async () => {
+          let currentTracesRows = tracesRows;
+          while (currentTracesRows.length === limit) {
+            const fromTraceIndex = currentTracesRows[limit - 1]!.traceIndex;
+            const fromBlock = Number(currentTracesRows[limit - 1]!.blockNumber);
+
+            const endClock = startClock();
+            currentTracesRows = await getEventBlockData_traces(
+              {
+                traceFilters,
+                transferFilters,
+                fromBlock,
+                fromTraceIndex,
+                toBlock: supremum,
+                chainId,
+                limit,
+              },
+              database,
+            );
+
+            common.metrics.ponder_database_method_duration.observe(
+              { method: "getEventBlockData_traces" },
+              endClock(),
+            );
+            for (const row of currentTracesRows) {
+              // @ts-ignore
+              tracesRows.push(row);
+            }
+          }
+        };
+
+        await Promise.all([
+          transactionsPromise,
+          transactionReceiptsPromise,
+          logsPromise,
+          tracesPromise,
+        ]);
```

Comment:
So these basically pull all the extra data from end of the limit to the supremum block?